lib/vrl/stdlib/src/encode_gzip.rs

```diff
@@ -0,0 +1,122 @@
+use ::value::Value;
+use flate2::read::GzEncoder;
+use nom::AsBytes;
+use std::io::Read;
+use vrl::prelude::expression::FunctionExpression;
+use vrl::prelude::*;
+
+fn encode_gzip(value: Value, compression_level: Option<Value>) -> Resolved {
+    let compression_level = match compression_level {
+        None => flate2::Compression::default(),
+        Some(value) => flate2::Compression::new(value.try_integer()? as u32),
+    };
+
+    let value = value.try_bytes()?;
+    let mut buf = Vec::new();
+    let _result = GzEncoder::new(value.as_bytes(), compression_level).read_to_end(&mut buf);
+
+    Ok(Value::Bytes(buf.into()))
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeGzip;
+
+impl Function for EncodeGzip {
+    fn identifier(&self) -> &'static str {
+        "encode_gzip"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "demo string",
+            source: r#"encode_base64(encode_gzip("encode_me"))"#,
+            result: Ok("H4sIAAAAAAAA/0vNS85PSY3PTQUAN7ZBnAkAAAA="),
+        }]
+    }
+
+    fn compile(
+        &self,
+        _state: &state::TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let compression_level = arguments.optional("compression_level");
+
+        Ok(EncodeGzipFn {
+            value,
+            compression_level,
+        }
+        .as_expr())
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "compression_level",
+                kind: kind::INTEGER,
+                required: false,
```

Comment:
It'd definitely add complexity but I wonder if we'd want to allow `best` and `fast` here as well - probably worth waiting until it's requested before adding, just came to mind.

[ref](https://cs.github.com/vectordotdev/vector/blob/9c5b0a137967986ccb11ee2675a7cbf4941398ad/src/sinks/util/buffer/compression.rs#L325-L331)